values/decoding/dictionary_based_string_iterator.go

```diff
@@ -65,5 +65,13 @@ func (it *dictionaryBasedStringIterator) Err() error {
 func (it *dictionaryBasedStringIterator) Close() {
 	it.extDict = nil
 	it.err = nil
+	// Close the underlying reader if it satisifies the `io.ReadCloser` iface.
+	rc, ok := it.reader.(io.ReadCloser)
```

Comment:
On a second thought, this is a bit worrisome compared to the approach where we implement a wrapper `SimpleReadCloser` that has a no op `Close` as this requires we understand where we need to cast and where otherwise.

Feels it's cleaner to define in `x/io/reader.go` 
```
import (
  "io"

  xio "github.com/m3db/m3x/io"
)

type SimpleReadCloser interface {
  io.Reader
  xio.SimpleCloser
}

type ReaderNoopCloser io.Reader

// Close is a no-op satistifying the `xio.SimpleCloser` interface.
func (r ReaderNoopCloser) Close() {}
```

Then implement this interface in `compressedReader` as well.

Finally pass `xio.SimpleReadCloser` to the raw/dictionary string iterator ctors (and where else closing reader is expected), casting `io.Reader` to `xio.ReaderNoopCloser` as needed.